includes/fbbot/class-omise-fbbot-endpoints.php

```diff
@@ -0,0 +1,141 @@
+<?php
+defined( 'ABSPATH' ) or die( "No direct script access allowed." );
+
+if ( class_exists( 'Omise_FBBot_Endpoints' ) ) {
+  return;
+}
+
+class Omise_FBBot_Endpoints extends WP_REST_Controller {
+
+	private static $instance;
+
+	/**
+	 * @var string
+	 */
+	private $facebook_page_verify_token;
+
+	private function __construct() {
+		$this->facebook_page_verify_token = Omise_FBBot_Configurator::get_fb_settings('facebook_page_verify_token');
+	}
+
+	public static function get_instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	public function register_bot_api_routes() {
+		$namespace = Omise_FBBot_Configurator::get_namespace();
+
+		register_rest_route( $namespace, '/webhook', array(
+            array(
+                'methods'  => WP_REST_Server::READABLE,
+                'callback' => array( $this, 'verify_fb_token_callback' ),
+                'permission_callback' => array( $this, 'verify_fb_token_request' )
+            ),
+
+            array(
+                'methods' => WP_REST_Server::CREATABLE,
+                'callback' => array($this, 'handle_request')
+            )
+        ) 
+    );
+
+    register_rest_route( $namespace, '/omise_triggered', array(
+            array(
+                'methods' => WP_REST_Server::CREATABLE,
+                'callback' => array( $this, 'triggered_from_omise' )
+            )
+        ) 
+    );
+
+    register_rest_route( $namespace, '/messenger_checkout', array(
+            array(
+                'methods' => WP_REST_Server::CREATABLE,
+                'callback' => array( $this, 'messenger_checkout' )
+            )
+        ) 
+    );
+
+    register_rest_route( $namespace, '/checking_payment', array(
+            array(
+                'methods' => WP_REST_Server::READABLE,
+                'callback' => array( $this, 'checking_payment' )
+            )
+        ) 
+    );
+	}
+
+	public function verify_fb_token_callback( $request ) {
+    $params = $request->get_query_params();
+    echo $params['hub_challenge'];
+    die();
+  }
+
+  public function verify_fb_token_request( $request = NULL ) {
+    $params = $request->get_query_params();
+
+    if ( $params && isset( $params['hub_challenge'] ) && $params['hub_verify_token'] == $this->facebook_page_verify_token ) {
+        return true;
+    }
+
+    return false;
+  }
+
+  public function handle_request ( $request ) {
```

Comment:
@igroomgrim kinda confuse on each of method names, do you think it would be better to add some prefix to indicate that these methods are about REST callback?

like, `callback_*`